lib/usage.ts

```diff
@@ -707,22 +707,31 @@ export function usage(yargs: YargsInstance, shim: PlatformShim) {
       descriptions,
     });
   };
-  self.unfreeze = function unfreeze() {
+  self.unfreeze = function unfreeze(defaultCommand = false) {
     const frozen = frozens.pop();
     // In the case of running a defaultCommand, we reset
     // usage early to ensure we receive the top level instructions.
     // unfreezing again should just be a noop:
     if (!frozen) return;
-    ({
-      failMessage,
-      failureOutput,
-      usages,
-      usageDisabled,
-      epilogs,
-      examples,
-      commands,
-      descriptions,
-    } = frozen);
+    // Addresses: https://github.com/yargs/yargs/issues/2030
+    if (defaultCommand) {
```

Comment:
I like that you just made this a boolean setting rather than adding a new method.